### PR TITLE
fix(task-status): Show Pending Tasks in Task Status Column

### PR DESCRIPTION
**Pull request by abhinandan13jan**
To fix bug ODC-1241:Show Pending Tasks in TaskStatus column.

### CODE REVIEW

File: frontend/packages/dev-console/src/components/pipelineruns/PipelineTaskStatus.tsx [hunk 1]
```diff
@@ -1,34 +1,39 @@
...
-} from '../../utils/pipeline-augment';
-import TaskStatusToolTip from './TaskStatusTooltip';
+import { Firehose } from '@console/internal/components/utils';
+import { PipelineRun } from '../../utils/pipeline-augment';
+import { PipelineModel } from '../../models';
+import { PipelineBars } from './PipelineBars';
 
 interface PipelineTaskStatusProps {
   pipelinerun: PipelineRun;
+  pipeline?: any;
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/pipelineruns/PipelineTaskStatus.tsx]: Do we have a type for a pipeline instead of using `any`?

Comment (abhinandan13jan) [reply frontend/packages/dev-console/src/components/pipelineruns/PipelineTaskStatus.tsx]: Updated to `K8sKind`

File: frontend/packages/dev-console/src/utils/pipeline-augment.ts [hunk 2]
```diff
@@ -186,31 +191,41 @@ export const getRunStatusColor = (status: string): StatusMessage => {
...
   const taskStatus: TaskStatus = {};
   Object.keys(runStatus).forEach((key) => {
     taskStatus[key] = 0;
   });
+  taskStatus[runStatus.Pending] =
+    pipelinerun &&
+    pipelinerun.status &&
+    pipelinerun.status.taskRuns &&
+    totalTasks >= Object.keys(pipelinerun.status.taskRuns).length
+      ? totalTasks - Object.keys(pipelinerun.status.taskRuns).length
```
[the remark this answers is not in the document]
Comment (abhinandan13jan) [reply frontend/packages/dev-console/src/utils/pipeline-augment.ts]: updated

File: frontend/packages/dev-console/src/utils/pipeline-augment.ts [hunk 3]
```diff
@@ -186,33 +191,42 @@ export const getRunStatusColor = (status: string): StatusMessage => {
...
     default:
       return { message: 'PipelineRun Not started Yet', pftoken: pendingColor };
   }
 };
 
-export const getTaskStatus = (pipelinerun: PipelineRun): TaskStatus => {
+export const getTaskStatus = (pipelinerun: PipelineRun, pipeline: Pipeline): TaskStatus => {
+  const totalTasks =
+    pipeline && pipeline.spec && pipeline.spec.tasks ? pipeline.spec.tasks.length : 0;
+  const plrTasks = Object.keys(pipelinerun.status.taskRuns);
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/src/utils/pipeline-augment.ts]:
```suggestion
  const plrTasks = pipelinerun && pipelinerun.status && pipelinerun.status.taskRuns ? Object.keys(pipelinerun.status.taskRuns) : [];
```

File: frontend/packages/dev-console/src/utils/pipeline-augment.ts [hunk 4]
```diff
@@ -186,33 +191,42 @@ export const getRunStatusColor = (status: string): StatusMessage => {
...
+  const plrTaskLength = plrTasks.length;
   const taskStatus: TaskStatus = {};
   Object.keys(runStatus).forEach((key) => {
     taskStatus[key] = 0;
   });
+  taskStatus[runStatus.Pending] =
+    pipelinerun && pipelinerun.status && pipelinerun.status.taskRuns && totalTasks >= plrTaskLength
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/src/utils/pipeline-augment.ts]:
```suggestion
    taskStatus[runStatus.Pending]  =  totalTasks >= plrTaskLength ? totalTasks - plrTaskLength : totalTasks;
```

**Comment by karthikjeeyar** [comment]
@abhinandan13jan Please resolve the conflicts and push again

File: frontend/packages/dev-console/src/utils/pipeline-augment.ts [hunk 5]
```diff
@@ -187,34 +191,43 @@ export const getRunStatusColor = (status: string): StatusMessage => {
...
+  const plrTasks =
+    pipelinerun && pipelinerun.status && pipelinerun.status.taskRuns
+      ? Object.keys(pipelinerun.status.taskRuns)
+      : [];
+  const plrTaskLength = plrTasks.length;
   const taskStatus: TaskStatus = {};
   Object.keys(runStatus).forEach((key) => {
     taskStatus[key] = 0;
   });
+  taskStatus[runStatus.Pending] = taskStatus[runStatus.Pending] =
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/src/utils/pipeline-augment.ts]: remove duplicate entry `taskStatus[runStatus.Pending] =`

File: frontend/packages/dev-console/src/components/pipelineruns/PipelineBars.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+import {
+  getTaskStatus,
+  runStatus,
+  getRunStatusColor,
+  Pipeline,
+  PipelineRun,
+} from '../../utils/pipeline-augment';
+import TaskStatusToolTip from './TaskStatusTooltip';
+
+interface PipelineBarProps {
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/src/components/pipelineruns/PipelineBars.tsx]: Export the prop types.

File: frontend/packages/dev-console/src/components/pipelineruns/PipelineTaskStatus.tsx [hunk 7]
```diff
@@ -1,34 +1,39 @@
...
-  runStatus,
-  getRunStatusColor,
-} from '../../utils/pipeline-augment';
-import TaskStatusToolTip from './TaskStatusTooltip';
+import { Firehose } from '@console/internal/components/utils';
+import { PipelineRun, Pipeline } from '../../utils/pipeline-augment';
+import { PipelineModel } from '../../models';
+import { PipelineBars } from './PipelineBars';
 
 interface PipelineTaskStatusProps {
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/src/components/pipelineruns/PipelineTaskStatus.tsx]: Export the prop types.

Comment (abhinandan13jan) [reply frontend/packages/dev-console/src/components/pipelineruns/PipelineTaskStatus.tsx]: alright

File: frontend/packages/dev-console/src/utils/pipeline-augment.ts (hunk 5 above)
Comment (abhinandan13jan) [reply frontend/packages/dev-console/src/utils/pipeline-augment.ts]: updated

File: frontend/packages/dev-console/src/components/pipelineruns/PipelineBars.tsx (hunk 6 above)
Comment (abhinandan13jan) [reply frontend/packages/dev-console/src/components/pipelineruns/PipelineBars.tsx]: updated

**Pull request merged**